extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java

```diff
@@ -658,13 +686,29 @@ public String validate(JwtContext jwtContext) throws MalformedClaimException {
                 if (!claims.hasClaim(claimName)) {
                     return "claim " + claimName + " is missing";
                 }
-                if (!claims.isClaimValueString(claimName)) {
-                    throw new MalformedClaimException("expected claim " + claimName + " to be a string");
-                }
-                var claimValue = claims.getStringClaimValue(claimName);
-                var targetValue = targetClaim.getValue();
-                if (!claimValue.equals(targetValue)) {
-                    return "claim " + claimName + " does not match expected value of " + targetValue;
+                Set<String> requiredClaimValues = targetClaim.getValue();
+                if (claims.isClaimValueString(claimName)) {
+                    if (requiredClaimValues.size() == 1) {
+                        String actualClaimValue = claims.getStringClaimValue(claimName);
+                        String requiredClaimValue = requiredClaimValues.iterator().next();
+                        if (!requiredClaimValue.equals(actualClaimValue)) {
+                            return "claim " + claimName + " does not match expected value of " + requiredClaimValues;
+                        }
+                    } else {
+                        throw new MalformedClaimException("expected claim " + claimName + " to be a list of strings");
```

Comment:
Propose to replace `to be a list of strings` with `must be a list of strings` here and in other cases